### PR TITLE
feat: Support non-default cluster domain

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Part of https://github.com/stackabletech/issues/issues/652

- [x] grepped for cluster.local and fixed
- [x] changelog (if needed)
- [x] values.yaml comment

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

File: deploy/helm/hdfs-operator/values.yaml [hunk 1]
```diff
@@ -47,3 +47,8 @@ nodeSelector: {}
 tolerations: []
 
 affinity: {}
+
+# When running on a non-default Kubernetes cluster domain and the auto detections is not working correctly,
```
Comment (maltesander) [inline deploy/helm/hdfs-operator/values.yaml]:
```suggestion
# When running on a non-default Kubernetes cluster domain and the auto detection is not working correctly,
```

**Review by maltesander — CHANGES_REQUESTED** [review]
LGTM otherwise!

File: deploy/helm/hdfs-operator/values.yaml [hunk 2]
```diff
@@ -47,3 +47,8 @@ nodeSelector: {}
 tolerations: []
 
 affinity: {}
+
+# When running on a non-default Kubernetes cluster domain and the auto detection is not working correctly,
+# you can set your custom cluster domain here.
+# See the https://docs.stackable.tech/home/stable/guides/kubernetes-cluster-domain guide for details
+# kubernetesClusterDomain: my-cluster.local
```
Comment (maltesander) [inline deploy/helm/hdfs-operator/values.yaml]:
```suggestion
# When running on a non-default Kubernetes cluster domain, the cluster domain can be configured here.
# See the https://docs.stackable.tech/home/stable/guides/kubernetes-cluster-domain guide for details.
# kubernetesClusterDomain: my-cluster.local
```

File: rust/operator-binary/src/config.rs [hunk 3]
```diff
@@ -183,7 +194,7 @@ impl HdfsSiteConfigBuilder {
                 format!("{}.{}.{}", address, self.logical_name, nn.pod_name),
                 format!(
                     "{}:{}",
-                    nn.fqdn(),
+                    nn.fqdn(cluster_info),
```
Comment (maltesander) [inline rust/operator-binary/src/config.rs]: Just a nit: lets do it consistent and use the identifiers instead of relying on order?

**Review by maltesander — CHANGES_REQUESTED** [review]

File: rust/operator-binary/src/config.rs (hunk 3 above)
Comment (sbernauer) [reply rust/operator-binary/src/config.rs]: Pls have a look at 09caef5

**Review by maltesander — APPROVED** [review]
LGTM!

**Pull request merged**